nmigen/hdl/rec.py

```diff
@@ -146,7 +143,18 @@ def concat(a, b):
                                                      src_loc_at=1 + src_loc_at)
 
     def __getattr__(self, name):
-        return self[name]
+        # must check `getattr` before `self` - we need to hit Value methods before fields
+        try:
+            # it seems we lose the implicit `self` argument at some point, so curry manually
+            value_attr = getattr(Value, name)
+            if callable(res):
```

Comment:
The sad part of this GitHub feature is that I think it only works one line at a time...
```suggestion
            if callable(value_attr):
```

Reply:
Yep, was just seeing that. I'll fix it in a proper commit...